common/cmake/devices.cmake

```diff
@@ -242,17 +242,23 @@ function(DEFINE_DEVICE_TYPE)
   #   DEVICE_TYPE <device_type>
   #   ARCH <arch>
   #   ARCH_XML <arch.xml>
+  #   [SCRIPT_OUTPUT_NAME]
+  #   [SCRIPTS]
   #   )
   # ~~~
   #
   # Defines a device type with the specified architecture.  ARCH_XML argument
   # must be a file target (see ADD_FILE_TARGET).
   #
+  # optional SCTIPTs can be run after the standard flow to augment the
+  # final arch xml. The name and script must be provided and each
+  # script will be run as `cmd input > output`
```

Comment:
Maybe consider running it as:

```
cmd < input > output
```

So that scripts don't need to be away of files at all, and just use stdin/stdout?